### DISCUSSION TITLE
Global path generation

### DISCUSSION THREAD

**Discussion by Achin17** [eleurent/highway-env#178] [Ideas]
Hi Eleurent! I was wondering if there is way, I can publish waypoints in terms of [x, y, theta] given a start and a goal position

**Comment by eleurent** [comment]
Hi @Achin17 
I did not understand what you mean exactly, but here are two remarks:

* If you want to be able to send target waypoints [x,y,theta] and have an agent learn to drive to this location, then it is the purpose of the [parking-v0](https://github.com/eleurent/highway-env#parking) environment, where waypoints are sampled randomly at the beginning of each episode (the waypoints also contain a target velocity, but it is always set to 0 in this example)
* If you want to use waypoints as continuous actions "go to that position" that the agent can execute (not learn) to achieve a high-level goal, then no environment exists at the moment for this specific task but you can easily add one if you wish. You will need to implement the position controller that will execute the high-level actions, but there already exists a heading and velocity controller so it should not be too hard.

Does that help? Feel free to elaborate if I misunderstood.
Edouard
